python/lsst/summit/utils/tmaUtils.py

```diff
@@ -0,0 +1,1346 @@
+    """Get the commands issued during an event.
+
+    Get the times at which the specified commands were issued during the event.
+
+    Parameters
+    ----------
+    client : `lsst_efd_client.efd_helper.EfdClient`
+        The EFD client to use.
+    event : `lsst.summit.utils.tmaUtils.TMAEvent`
+        The event to plot.
+    commands : `list` of `str`, optional
+        The commands or command aliases to look for. Defaults to
+        ['raDecTarget'].
+    log : `logging.Logger`, optional
+        The logger to use. If not specified, a new logger will be created if
+        needed.
+    doLog : `bool`, optional
+        Whether to log messages. Defaults to True.
+
+    Returns
+    -------
+    commands : `dict` of `str` : `astropy.time.Time`
+        A dictionary of the commands and the times at which they were issued.
+    """
+    # TODO: Add support for padding the event here to allow looking for
+    # triggering commands before the event
+
+    # TODO: Change this to always return a list of times, and remove warning
+    # about finding multiple commands. Remember to update docs and plotting
+    # code.
+    if log is None and doLog:
+        log = logging.getLogger(__name__)
+
+    commands = ensure_iterable(commands)
+    fullCommands = [c if c not in COMMAND_ALIASES else COMMAND_ALIASES[c] for c in commands]
+    del commands  # make sure we always use their full names
+
+    ret = {}
+    for command in fullCommands:
+        data = getEfdData(client, command, event=event, noWarn=True)
+        if data.empty:
+            if doLog:
+                log.info(f'Found no command issued for {command} during event')
+            ret[command] = None
+        elif len(data) > 1:
+            if doLog:
+                log.warning(f'Found multiple commands issued for {command} during event, returning None')
+            ret[command] = None
+        else:
+            assert len(data) == 1  # this must be true now
+            commandTime = data.private_sndStamp
+            ret[command] = Time(commandTime, format='unix_tai')
+
+    return ret
+
+
+def _initializeTma(tma):
+    """Helper function to turn a TMA into a valid state for testing.
+
+    Do not call directly in normal usage or code, as this just arbitrarily
+    sets values to make the TMA valid.
+    """
+    tma._parts['azimuthInPosition'] = False
+    tma._parts['azimuthMotionState'] = AxisMotionState.STOPPED
+    tma._parts['azimuthSystemState'] = PowerState.ON
+    tma._parts['elevationInPosition'] = False
+    tma._parts['elevationMotionState'] = AxisMotionState.STOPPED
+    tma._parts['elevationSystemState'] = PowerState.ON
+
+
+def _turnOn(tma):
+    """Helper function to turn TMA axes on for testing.
+
+    Do not call directly in normal usage or code, as this just arbitrarily
+    sets values to turn the axes on.
+    """
+    tma._parts['azimuthSystemState'] = PowerState.ON
+    tma._parts['elevationSystemState'] = PowerState.ON
```

Comment:
Thanks, I've moved `_turnOn()` into the tests themselves as this really isn't needed anywhere else at all, and I've left `_initializeTma()` in there as an _underMethod, because I do actually (for now) need to use it in that code itself, not just tests, but people should absolutely not be calling it elsewhere.